File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    
    for(let i = 0; i < progresses.length; i++){
        const a = progresses[i];
        
        progresses[i] = Math.floor((100 - a) / speeds[i]);
        if((100 - a) % speeds[i]) progresses[i] += 1;
    }
    
    while(progresses.length){
        const first = progresses.shift();
        var count = 1;
        
        while(progresses.length && progresses[0] <= first){
            progresses.shift();
            count++;
        }
        answer.push(count);
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []

    current_release_day = math.ceil((100 - progresses[0]) / speeds[0])
    count = 1

    for progress, speed in zip(progresses[1:], speeds[1:]):
        day = math.ceil((100 - progress) / speed)

        if day <= current_release_day:
            count += 1
        else:
            answer.append(count)
            current_release_day = day
            count = 1

    answer.append(count)
    return answer